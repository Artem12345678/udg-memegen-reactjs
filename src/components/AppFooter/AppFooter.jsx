import React from "react";
import Nav from "react-bootstrap/Nav";
import { Link } from "react-router-dom";
import SocialList from "../SocialList/SocialList";

import "./AppFooter.scss";

const AppFooter = () => {
  return (
    <footer className="footer bg-secondary">
      <div className="container">
        <div className="footer__nav">
          <Nav>
            <Nav.Link as={Link} to="/" className="nav-link--footer">
              Home
            </Nav.Link>
          </Nav>

          <div className="footer__social">
            <SocialList />
          </div>
        </div>
      </div>
    </footer>
  );
};

export default AppFooter;
