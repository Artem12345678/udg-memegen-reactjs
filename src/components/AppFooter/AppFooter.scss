.footer {
  padding: 30px 0;

  &__nav {
    display: flex;
    justify-content: space-between;
  }
}

.nav-link {
  &--footer {
    color: #fff;

    &:hover,
    &:focus,
    &:active {
      color: darken(#fff, 10%);
    }
  }
}
