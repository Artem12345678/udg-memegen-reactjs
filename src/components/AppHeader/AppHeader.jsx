import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import { Link } from "react-router-dom";

import "./AppHeader.scss";

const AppHeader = () => {
  return (
    <header className="header bg-primary">
      <div className="container">
        <Navbar variant="dark" expand="lg">
          <Navbar.Brand as={Link} to="/" className="brand">
            MemeGen
          </Navbar.Brand>

          <Navbar.Toggle aria-controls="nav-collapse" />

          <Navbar.Collapse id="nav-collapse">
            <Nav className="ml-auto">
              <Nav.Link as={Link} to="/">
                Home
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      </div>
    </header>
  );
};

export default AppHeader;
