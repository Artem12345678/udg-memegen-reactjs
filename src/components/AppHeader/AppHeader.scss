.header {
  padding: 10px 0;
}

.brand {
  font-size: 56px !important;
  font-weight: 700 !important;
  display: block;
  letter-spacing: -1.4px !important;
  color: #fff !important;
}
