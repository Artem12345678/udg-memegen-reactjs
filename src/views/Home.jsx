import React from "react";
import MemeGen from "../components/MemeGen/MemeGen";

const Home = () => {
  return (
    <div className="home">
      <MemeGen />
    </div>
  );
};

export default Home;
