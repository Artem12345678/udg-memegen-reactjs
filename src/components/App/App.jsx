import React from "react";
import { BrowserRouter } from "react-router-dom";

import AppHeader from "../AppHeader/AppHeader";
import AppFooter from "../AppFooter/AppFooter";
import Router from "../../router";

import "./App.scss";

const App = () => {
  return (
    <div id="app">
      <BrowserRouter>
        <AppHeader />

        <main className="container">
          <Router />
        </main>

        <AppFooter />
      </BrowserRouter>
    </div>
  );
};

export default App;
