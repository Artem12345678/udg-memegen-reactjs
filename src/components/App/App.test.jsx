import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import App from "./App";
import AppHeader from "../AppHeader/AppHeader";
import AppFooter from "../AppFooter/AppFooter";
import Home from "../../views/Home";

describe("App.jsx", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders <AppHeader/> component", () => {
    expect(wrapper.find(AppHeader).exists()).toBe(true);
  });

  it("renders <AppFooter/> component", () => {
    expect(wrapper.find(AppFooter).exists()).toBe(true);
  });

  describe("with router", () => {
    it("renders <Home/> component by default", () => {
      wrapper = mount(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      );

      expect(wrapper.find(Home)).toHaveLength(1);
    });

    it('renders <Home/> component for "/" route path', () => {
      wrapper = mount(
        <MemoryRouter initialEntries={["/"]}>
          <App />
        </MemoryRouter>
      );

      expect(wrapper.find(Home)).toHaveLength(1);
    });
  });
});
