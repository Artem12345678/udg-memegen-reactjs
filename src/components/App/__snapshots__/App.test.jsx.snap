// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx matches snapshot 1`] = `
<div
  id="app"
>
  <BrowserRouter>
    <AppHeader />
    <main
      className="container"
    >
      <Router />
    </main>
    <AppFooter />
  </BrowserRouter>
</div>
`;
